src/GuessControl.js

```diff
@@ -1,47 +1,28 @@
-import React, { Component } from "react";
-import Button from "./Button";
+import { useState } from "react"
+import Button from "./Button"
 
-class GuessControl extends Component {
-  constructor(props) {
-    super(props);
+function GuessControl(props) {
+  const [currentGuess, setCurrentGuess] = useState("")
 
-    this.state = {
-      currentGuess: "",
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleInputChange = this.handleInputChange.bind(this);
-    this.onSubmitGuess = this.onSubmitGuess.bind(this);
-  }
-
-  handleInputChange(event) {
-    this.setState({ currentGuess: event.target.value });
+  function handleInputChange(event) {
+    setCurrentGuess(event.target.value)
   }
 
-  onSubmitGuess() {
+  function onSubmitGuess() {
     // Since the values from an HTML input are strings by default,
     //  convert to a number for the returned guess value
     //  by passing in the string to the Number function.
     // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
-    this.props.onGuess(Number(this.state.currentGuess));
-    this.setState({ currentGuess: "" });
+    props.onGuess(Number(currentGuess))
```

Comment:
then this would be `onGuess` instead of `props.onGuess`. Idk developers are lazy lol The less typing the better. Which is one of the reason why developers chose functions over class components 😆